mypy/typestate.py

```diff
@@ -29,6 +33,26 @@ class TypeState:
     _subtype_caches = {}  # type: ClassVar[Dict[TypeInfo, Set[Tuple[Instance, Instance]]]]
     _subtype_caches_proper = {}  # type: ClassVar[Dict[TypeInfo, Set[Tuple[Instance, Instance]]]]
 
+    # This contains protocol dependencies generated after running a full build,
+    # or after an update. These dependencies are special because:
+    #   * They are a global property of the program; i.e. some dependencies for imported
+    #     classes can be generated in the importing modules.
+    #   * Because of the above, they are serialized separately, after a full run,
+    #     or a full update.
+    # `proto_deps` can be None if after deserialization it turns out that they are
+    # inconsistent with the other cache files (or an error occurred during deserialization).
+    # A blocking error will be generated in this case, since we can't proceed safely.
+    proto_deps = {}  # type: Optional[Dict[str, Set[str]]]
+
+    # Protocols (full names) a given class attempted to implement.
+    # Used to calculate fine grained protocol dependencies and optimize protocol
+    # subtype cache invalidation in fine grained mode.
+    _attempted_protocols = {}  # type: Dict[TypeInfo, Set[str]]
+    # We also snapshot protocol members of the above protocols.
+    _checked_against_members = {}  # type: Dict[TypeInfo, Set[str]]
+    # TypeInfos that has been reprocessed since latest dependency snapshot update.
+    _reprocessed_types = set()  # type: Set[TypeInfo]
```

Comment:
The term 'reprocessed' is a bit unclear here. Can you spell out what it means exactly? Does it only apply to fine-grained incremental mode?

Reply:
> Does it only apply to fine-grained incremental mode?

Essentially yes, this is is an optimization to avoid gathering dependencies from all registered types, but only from those re-checked during last update. For full build, this simply contains all registered types, we can't optimize this.